src/IssueViz/Editor/LineHashCalculator.cs

```diff
@@ -52,27 +48,25 @@ internal LineHashCalculator(IChecksumCalculator checksumCalculator)
             this.checksumCalculator = checksumCalculator;
         }
 
-        public string Calculate(string documentText, int oneBasedLineNumber)
+        public string Calculate(ITextDocument documentText, int oneBasedLineNumber)
         {
-            if (string.IsNullOrEmpty(documentText) || oneBasedLineNumber < 1)
+            if (oneBasedLineNumber < 1 || documentText?.TextBuffer?.CurrentSnapshot == null)
             {
                 return null;
             }
 
-            var lines = GetLines(documentText);
+            // SonarLint issues line numbers are 1-based, span lines are 0-based
+            var lineToHash = documentText.TextBuffer.CurrentSnapshot.GetLineFromLineNumber(oneBasedLineNumber - 1);
```

Comment:
Might be worth commenting that we're letting the text document find the line as it is aware of different types of line break.